### PR TITLE
Add support for Async::HTTP::Client

### PR DESCRIPTION
**Pull request by ayanko**
Add support for new asynchronous http library.
See https://github.com/socketry/async-http

### CODE REVIEW

**Comment by bblimke** [comment]
@ayanko thank you for the PR. It will be great to support Async::HTTP in Webmock.

I'd like to make sure that the adapter is compatible with all WebMock usecases. Perhaps I have missed that, but I don't see `include_examples "with WebMock"` anywhere is the spec.

Passing `"with WebMock"` examples is required for all http client adapters to guarantee sure they are fully compatible with WebMock API.

I guess that some AsyncHttpRequestSpecHelper will be needed as well for "with WebMock" compatibility.

**Comment by ayanko** [comment]
@bblimke Updated

* Added spec helper
*  Introduced`:no_content_length_header` option
* Added global shared examples

**Comment by ayanko** [comment]
@bblimke 

`Async::HTTP::Client` deletes `Content-Length` header from server response so I added :no_content_length_header option to shared specs.


See https://github.com/socketry/protocol-http1/blob/master/lib/protocol/http1/connection.rb#L441 for details

File: spec/acceptance/async_http_client/async_http_client_spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          )
+        end
+      rescue Exception => e
+        e
+      end
+    end.wait
+  end
+
+  def client_timeout_exception_class
+    "WebMock timeout error"
```
Comment (bblimke) [inline spec/acceptance/async_http_client/async_http_client_spec_helper.rb]: @ayanko shouldn't that be Async::TimeoutError?

Comment (ayanko) [reply spec/acceptance/async_http_client/async_http_client_spec_helper.rb]: @bblimke Yeap. Updated.

**Comment by ayanko** [comment]
@bblimke What else?

**Comment by bblimke** [comment]
@ayanko I believe it's all good. Thank you.

**Comment by bblimke** [comment]
Released as 3.7.0

**Pull request merged**